lib/logstash/filters/split.rb

```diff
@@ -66,6 +66,8 @@ def register
   public
   def filter(event)
     
+    # Convert field to a string based on any format values    
+    @field = event.sprintf(@field)
```

Comment:
What is the issue for save it back to @field ? I don't understand why I can't do that :)
On the first change, I replace the next statement but "@field" is also used few lines after and it needs to keep the new field version, that's why I change it directly and store it into @field